lib/loaders/base.rb

```diff
@@ -8,12 +8,41 @@ def self.load(path)
       new.load(path)
     end
 
-    def initialize(path)
+    def initialize(path, chunker: nil, **kwargs)
       @path = path
+      @chunker = setup_chunker(chunker)
     end
 
     def loadable?
       raise NotImplementedError
     end
+
+    def load
+      raise NotImplementedError
+    end
+
+    def load_chunked
```

Comment:
Should this actually be a part of a loader? It seems like `load_chunked` is pretty specific use case when you want your data to be chunked. I'd keep loader's API simple (actual implementation would be complex, though) and just give an API to chunk data before loading it to vector database.

Reply:
Thanks for highlighting this. I think what you propose is possible and I like the folllowing the most.

```ruby
loader = Loaders::Pdf.new('./1.pdf')
data = loader.load
SomeClass.add_data data.chunked
```

This would require `loader.load` to return an object that is chunkable. Let me play with this some more or have a chat with @andreibondarev about flexible paths forward.

Reply:
@rickychilcott @alchaplinsky What do you guys think about the following interface:
```
loader = Loaders::Pdf.new('./1.pdf')

data = loader.load(delimeter:, delimeter_function:, chunk_size:) # ... any other chunking-related params
# or no chunking
data = loader.load

SomeClass.add_data data
```